### PR TITLE
Improves Error Reporting

### PR DESCRIPTION
**Pull request by djblue**
@Bdthomson 
@andrewkfiedler

### CODE REVIEW

File: test/ace.spec.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+const { promisify } = require('util')
+const fs = require('fs')
+
+const stat = promisify(fs.stat)
+const access = promisify(fs.access)
+const writeFile = promisify(fs.writeFile)
+const rimraf = promisify(require('rimraf'))
+const mkdirp = promisify(require('mkdirp'))
+
+const assertFile = async path => {
```
Comment (Bdthomson) [inline test/ace.spec.js]: Not used.

File: test/ace.spec.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+const resolve = (...args) => {
+  return path.resolve(__dirname, 'example-project', ...args)
+}
+
+const ace = (...args) =>
+  new Promise(done => {
+    const cwd = resolve()
+    const stdio = 'ignore'
+    //const stdio = 'inherit'
```
Comment (Bdthomson) [inline test/ace.spec.js]: comment

File: lib/mocha-headless-chrome.js [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+    const result = puppeteer.launch(options).then(browser =>
+      browser
+        .pages()
+        .then(pages => pages.pop())
+        .then(configureViewport.bind(this, width, height))
+        .then(page => {
+          page.on('console', handleConsole)
+          page.on('dialog', dialog => dialog.dismiss())
+          //page.on('pageerror', err => console.error(err));
```
Comment (Lambeaux) [inline lib/mocha-headless-chrome.js]: ❓ Is this still needed?

File: lib/mocha-headless-chrome.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            )
+            .then(() => page.evaluate(() => window.__mochaResult__))
+            .then(obj => {
+              browser.close()
+              return obj
+            })
+        })
+    )
+
+    resolve(result)
```
Comment (Lambeaux) [inline lib/mocha-headless-chrome.js]: ❓ Would resolving the promise ensure that, beyond this point, the data is accessible? Is this a synchronization point?

File: test/ace.spec.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+          license: 'MIT',
+          main: 'src/main/webapp/index.js',
+          'context-path': '/my-app',
+        },
+        null,
+        2
+      ),
+      'pom.xml': '',
+      'src/main/webapp/index.js': `module.exports = () => 'hello, world'`,
+      'src/main/webapp/example.spec.js': `it('should pass', () => {})`,
```
Comment (Lambeaux) [inline test/ace.spec.js]: ❓ Is there a way to get tooling on JavaScript like this? For instance, providing raw JavaScript to a function that will simply stringify it?

**Pull request merged**